src/schema/v2/Alerts/updateAlertMutation.ts

```diff
@@ -0,0 +1,182 @@
+import {
+  GraphQLObjectType,
+  GraphQLUnionType,
+  GraphQLNonNull,
+  GraphQLString,
+  GraphQLBoolean,
+  GraphQLList,
+} from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import {
+  GravityMutationErrorType,
+  formatGravityError,
+} from "lib/gravityErrorHandler"
+import { ResolverContext } from "types/graphql"
+import { AlertSettingsInputType, AlertType } from "./"
+
+const SuccessType = new GraphQLObjectType<any, ResolverContext>({
+  name: "UpdateAlertSuccess",
+  isTypeOf: (data) => data.id,
+  fields: () => ({
+    alert: {
+      type: AlertType,
+      resolve: async ({ search_criteria, id, ...rest }) => {
+        return {
+          ...search_criteria,
+          id, // Inject the ID from the `UserSearchCriteria` object
+          settings: rest,
+        }
+      },
+    },
+  }),
+})
+
+const ErrorType = new GraphQLObjectType<any, ResolverContext>({
+  name: "UpdateAlertFailure",
+  isTypeOf: (data) => {
+    return data._type === "GravityMutationError"
+  },
+  fields: () => ({
+    mutationError: {
+      type: GravityMutationErrorType,
+      resolve: (err) => err,
+    },
+  }),
+})
+
+const ResponseOrErrorType = new GraphQLUnionType({
+  name: "UpdateAlertResponseOrError",
+  types: [SuccessType, ErrorType],
+})
+
+export const updateAlertMutation = mutationWithClientMutationId<
+  any,
+  any,
+  ResolverContext
+>({
+  name: "updateAlert",
+  description: "Create an alert",
+  inputFields: {
+    acquireable: {
```

Comment:
Can we share these input fields with the above?

Reply:
They're slightly different  - in the create, `artistIDs` is required, and in the update, `id` is required. I don't find it _too_ objectionable to just list them inline, but can also move the fields in common to a shared location.

Reply:
Ah, was just thinking that they might be identical, but in any case it is a lot of args, and most are shared. Its nbd tho... Less than 3 dupes

Reply:
I went for the update, I like it!